tests/test_diagnostics.py

```
@@ -0,0 +1,244 @@
+
+    assert out["present"] is True
+    assert out["runtime_type"] == "mapping"
+    assert out["runtime_keys"] == ["aa...zz"]
+
+
+@pytest.mark.asyncio
+async def test_async_get_config_entry_diagnostics_redacts_sensitive_fields(
+    mock_hass, monkeypatch
+):
+    redact_key = next(iter(TO_REDACT))
+    secret = "just_a_test_value"  # nosec B105
```

Ruff (0.14.10)

Possible hardcoded password assigned to: "secret"

(S105)